src/frontend/components/JoinDiscussionLeaveButton/index.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { useState } from 'react';
+import { defineMessages, FormattedMessage } from 'react-intl';
+import { Redirect } from 'react-router-dom';
+
+import { DashboardButton } from '../DashboardPaneButtons/DashboardButtons';
+import { converse } from '../../utils/window';
+import { PLAYER_ROUTE } from '../routes';
+import { modelName } from '../../types/models';
+import { useParticipantWorkflow } from '../../data/stores/useParticipantWorkflow';
+
+const messages = defineMessages({
+  leaveDiscussion: {
+    defaultMessage: 'Leave the discussion',
+    description: 'Leave the discussion',
+    id: 'components.JoinDiscussionLeaveButton.leaveDiscussion',
+  },
+});
+
+export const JoinDiscussionLeaveButton = () => {
+  const [leaved, setLeaved] = useState(false);
+  const reset = useParticipantWorkflow((state) => state.reset);
+
+  const onClick = () => {
+    converse.participantLeaves();
+    reset();
+    setLeaved(true);
+  };
+
+  if (leaved) {
+    return <Redirect to={PLAYER_ROUTE(modelName.VIDEOS)} />;
+  }
```

Comment:
You could use `withRouter` and redirect through `props.history.push(PLAYER_ROUTE(modelName.VIDEOS)`. In this way you could remove the use of a state.